application/android/app/src/main/java/com/group12/beabee/views/entities/TaskCardViewAdapter.java

```diff
@@ -0,0 +1,75 @@
+package com.group12.beabee.views.entities;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.group12.beabee.R;
+import com.group12.beabee.models.TaskShort;
+
+import java.util.List;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+public class TaskCardViewAdapter extends RecyclerView.Adapter<TaskCardViewAdapter.ViewHolder> {
+
+
+    private List<TaskShort> taskShortList;
+    private IOnTaskClickedListener onItemClickedListener;
+
+    public void setData(List<TaskShort> taskShorts){
+        taskShortList = taskShorts;
+        notifyDataSetChanged();
+    }
+
+    public void setItemClickListener(IOnTaskClickedListener listener){
+        onItemClickedListener = listener;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_task_cardview, parent, false);
+        return new ViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
+        holder.BindData(taskShortList.get(position));
+    }
+
+    @Override
+    public int getItemCount() {
+
+        return taskShortList.size();
```

Comment:
As I experienced similarly this piece of code may cause a null pointer exception because the data list might be null in the view creation. Instead we should check whether the list is null and return 0 if so.